MAINT: Update NEP template.

hameerabbasi opened the pull request:
<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message
-->

hameerabbasi committed:
MAINT: Update NEP template.

mattip commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
 This section describes the need for the NEP.  It should describe the existing
 problem that it is trying to solve and why this NEP makes the situation better.
 It should include examples of how the new functionality would be used and
 perhaps some use cases.
```
Maybe add something about intended users "It should describe the existing problem, who it affects, what it is trying to solve, and why ..."  and end with "would be used, intended use cases, and pseudo-code describing its use"?

hameerabbasi commented:
cc @rgommers

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
+This section should list relevant and/or similar technologies, possibly in other
+libraries. It does not need to be comprehensive, just list the major examples of
+prior and relevant art.
 
```
I think "Detailed description" should be kept here. And pseudo-code / examples should stay here.

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
-It should include examples of how the new functionality would be used and
-perhaps some use cases.
+problem, who it affects, what it is trying to solve, and why.
+It should include examples of how the new functionality would be used, intended
```
suggest to move this sentence to "Detailed description" further down.

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
-problem that it is trying to solve and why this NEP makes the situation better.
-It should include examples of how the new functionality would be used and
-perhaps some use cases.
+problem, who it affects, what it is trying to solve, and why.
```
After this, I suggest: "This section should explicitly address the scope and requirements of the proposed change."

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
-
-Detailed description
---------------------
+Motivation
```
"Motivation" is an improvement. I suggest "Motivation and scope"

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,20 @@ Abstract
+It should include examples of how the new functionality would be used, intended
+use-cases and psuedo-code describing its use.
 
+Literature Review
```
I think this is a good section to have, most NEPs do have relevant prior art somewhere. The title isn't optimal though, since it's mostly not literature (that gives the wrong signal to an academic) - how about "Related work"?

rgommers reviewed (dismissed):
Thanks, this is an improvement. I think while some of the questions were in the text before, they usually weren't answered well (or at all). Separating these questions from the detailed description of the design should help.

hameerabbasi committed:
Address feedback by @rgommers.

hameerabbasi commented:
@rgommers @mattip Anything else needed here?

rgommers commented on doc/neps/nep-template.rst:
```diff
@@ -13,15 +13,27 @@ Abstract
+Motivation and Scope
+--------------------
+
+This section describes the need for the NEP.  It should describe the existing
```
This is now duplicate with "detailed description". The first two sentences of that section should be replaced. I can have a go at it.

mattip replied:
@hameerabbasi @rgommers ping

rgommers committed:
DOC: minor tweak to content and formatting of NEP template.

Removes a duplicate sentence and overlap between the "Motivation and Scope"
and "Detailed implementation" sections.

mattip commented:
Thanks @hameerabbasi. Additional improvements can be done in further pull requests as needed.
